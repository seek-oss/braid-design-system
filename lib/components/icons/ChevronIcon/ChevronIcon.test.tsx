import React from 'react';
import { ChevronIcon } from '../..';
import { render } from 'react-testing-library';
import forEachTheme from '../../../test/utils/forEachTheme';

describe('ChevronIcon', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('Default', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Sized', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon size="large" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Inline', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon inline />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Up', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon direction="up" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Down', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon direction="down" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Left', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon direction="left" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Right', () => {
      const { container } = render(
        <ThemeProvider>
          <ChevronIcon direction="right" />
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
