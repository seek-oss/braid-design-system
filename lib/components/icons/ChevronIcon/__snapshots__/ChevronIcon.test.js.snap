// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronIcon Theme: jobStreet Default 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Down 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Inline 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="14"
  viewBox="0 0 1024 1024"
  width="14"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Left 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__left Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Right 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__right Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Sized 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: jobStreet Up 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__up Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Default 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Down 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Inline 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="16"
  viewBox="0 0 1024 1024"
  width="16"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Left 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__left Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Right 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__right Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Sized 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAnz Up 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__up Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Default 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Down 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Inline 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="16"
  viewBox="0 0 1024 1024"
  width="16"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Left 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__left Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Right 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__right Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Sized 1`] = `
<svg
  class="ChevronIcon__root Box__root borderColor__default"
  height="24"
  viewBox="0 0 1024 1024"
  width="24"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;

exports[`ChevronIcon Theme: seekAsia Up 1`] = `
<svg
  class="ChevronIcon__root ChevronIcon__up Box__root borderColor__default"
  height="20"
  viewBox="0 0 1024 1024"
  width="20"
>
  <path
    d="M945 305l78 67-510 524-510-524 75-69 435 451 432-449z"
  />
</svg>
`;
